src/coreclr/jit/emitarm64.cpp

```diff
@@ -1124,6 +1124,17 @@ void emitter::emitInsSanityCheck(instrDesc* id)
             assert(id->idInsOpt() == INS_OPTS_SCALABLE_H);
             break;
 
+        case IF_SVE_GD_2A: // .........x.xx... ......nnnnnddddd -- SVE2 saturating extract narrow
+            elemsize = id->idOpSize();
+            assert(insOptsScalable(id->idInsOpt()));
+            assert(isVectorRegister(id->idReg1())); // nnnnn
+            assert(isVectorRegister(id->idReg2())); // ddddd
+            assert(optGetSveElemsize(id->idInsOpt()) != EA_8BYTE);
```

Comment:
`assert(insOptsScalableFloat(id->idInsOpt()));` is used elsewhere and is essentially the same check. 

But, I _might_ prefer this way. We could then get rid of `insOptsScalableWords()`, `insOptsScalableAtLeastHalf()` and `insOptsScalableFloat()`